python/pyspark/sql/context.py

```diff
@@ -315,28 +335,19 @@ def createDataFrame(self, data, schema=None, samplingRatio=None):
         else:
             rdd = data
 
-        if schema is None:
-            schema = self._inferSchema(rdd, samplingRatio)
+        if not isinstance(schema, StructType):
```

Comment:
Actually we are doing more than that. Let's change it so if it is the wrong type, we will throw typerror.